worker/migrationmaster/worker.go

```diff
@@ -333,33 +334,52 @@ func (w *Worker) doQUIESCE(status coremigration.MigrationStatus) (coremigration.
 }
 
 func (w *Worker) prechecks(status coremigration.MigrationStatus) error {
-	w.setInfoStatus("performing source prechecks")
-	err := w.config.Facade.Prechecks()
-	if err != nil {
-		return errors.Annotate(err, "source prechecks failed")
-	}
-
-	w.setInfoStatus("performing target prechecks")
 	model, err := w.config.Facade.ModelInfo()
 	if err != nil {
 		return errors.Annotate(err, "failed to obtain model info during prechecks")
 	}
-	conn, err := w.openAPIConn(status.TargetInfo)
+	targetConn, err := w.openAPIConn(status.TargetInfo)
 	if err != nil {
 		return errors.Annotate(err, "failed to connect to target controller during prechecks")
 	}
-	defer conn.Close()
+	defer targetConn.Close()
 
-	if conn.ControllerTag() != status.TargetInfo.ControllerTag {
-		return errors.Errorf("unexpected target controller UUID (got %s, expected %s)",
-			conn.ControllerTag(), status.TargetInfo.ControllerTag)
+	targetControllerVersion, err := w.getTargetControllerVersion(targetConn)
+	if err != nil {
+		return errors.Annotate(err, "cannot get target controller version")
 	}
 
-	targetClient := migrationtarget.NewClient(conn)
+	w.setInfoStatus("performing source prechecks")
+	err = w.config.Facade.Prechecks(targetControllerVersion)
+	if err != nil {
+		return errors.Annotate(err, "source prechecks failed")
+	}
+
+	w.setInfoStatus("performing target prechecks")
+	if targetConn.ControllerTag() != status.TargetInfo.ControllerTag {
+		return errors.Errorf("unexpected target controller UUID (got %s, expected %s)",
+			targetConn.ControllerTag(), status.TargetInfo.ControllerTag)
+	}
+	targetClient := migrationtarget.NewClient(targetConn)
 	err = targetClient.Prechecks(model)
 	return errors.Annotate(err, "target prechecks failed")
 }
 
+func (w *Worker) getTargetControllerVersion(conn api.Connection) (version.Number, error) {
+	// Use target Connection to access the ModelInfo via MigrationMaster API.
+	// This is a workaround.
+	client := controller.NewClient(conn)
```

Comment:
I don't think we can easily do that here. Because we are creating a target controller client in the source controller worker. You can see we have a few places doing this in this worker, for example, https://github.com/juju/juju/blob/2.9/worker/migrationmaster/worker.go#L358